File: src/GoonEngine/ecs/components/locationComponent.c
#include <GoonEngine/gnpch.h>
#include <GoonEngine/ecs/components/locationComponent.h>

LocationComponent* gnLocationComponentNew()
{
    LocationComponent *locComponent = calloc(1, sizeof(*locComponent));
    return locComponent;
}
int gnLocationComponentGetX(LocationComponent* component)
{
    return component->x;
}
int gnLocationComponentGetY(LocationComponent* component)
{
    return component->y;
}
void gnLocationComponentSetX(LocationComponent* component, int x)
{
    component->x = x;
}
void gnLocationComponentSetY(LocationComponent* component, int y)
{
    component->y = y;
}
void gnLocationComponentSetXY(LocationComponent* component, int x, int y)
{
    component->x = x;
    component->y = y;
}

File: GoonWorld/GoonEngine/components/LocationComponent.cs
namespace GoonEngine;
public class LocationComponent : Component
{
    public LocationComponent(int x = 0, int y = 0)
    {
        _componentDataPointer = Api.Components.LocationComponent.gnLocationComponentNew();
        Location.x = x;
        Location.y = y;
        _ecsComponentDataPtr = ECS.NewComponent(ComponentTypes.LOCATION_COMPONENT, ComponentPtr );
    }

    public unsafe LocationComponent()
    {
        _componentDataPointer = Api.Components.LocationComponent.gnLocationComponentNew();
        _ecsComponentDataPtr = ECS.NewComponent(ComponentTypes.LOCATION_COMPONENT, ComponentPtr );
    }

    public unsafe ref Models.LocationComponent Location => ref *(Models.LocationComponent*)_componentDataPointer;
}